src/main/java/de/tum/in/www1/artemis/repository/LearningGoalProgressRepository.java

```diff
@@ -0,0 +1,75 @@
+package de.tum.in.www1.artemis.repository;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import de.tum.in.www1.artemis.domain.LearningGoalProgress;
+
+@Repository
+public interface LearningGoalProgressRepository extends JpaRepository<LearningGoalProgress, Long> {
+
+    @Transactional
+    @Modifying
+    void deleteAllByLearningGoalId(Long learningGoalId);
+
+    @Transactional
+    @Modifying
+    void deleteAllByUserId(Long userId);
+
+    @Query("""
+            SELECT learningGoalProgress
+            FROM LearningGoalProgress learningGoalProgress
+            WHERE learningGoalProgress.learningGoal.id = :learningGoalId
+            """)
+    List<LearningGoalProgress> findAllByLearningGoalId(@Param("learningGoalId") Long learningGoalId);
+
+    @Query("""
+            SELECT learningGoalProgress
+            FROM LearningGoalProgress learningGoalProgress
+            WHERE learningGoalProgress.learningGoal.id = :learningGoalId
+            AND learningGoalProgress.user.id = :userId
+            """)
+    Optional<LearningGoalProgress> findByLearningGoalIdAndUserId(@Param("learningGoalId") Long learningGoalId, @Param("userId") Long userId);
+
+    @Query("""
+            SELECT learningGoalProgress
+            FROM LearningGoalProgress learningGoalProgress
+            LEFT JOIN FETCH learningGoalProgress.user
+            LEFT JOIN FETCH learningGoalProgress.learningGoal
+            WHERE learningGoalProgress.learningGoal.id = :learningGoalId
+            AND learningGoalProgress.user.id = :userId
+            """)
```

Comment:
```suggestion
    @Query("""
            SELECT p
            FROM LearningGoalProgress p
                LEFT JOIN FETCH p.user
                LEFT JOIN FETCH p.learningGoal
            WHERE p.learningGoal.id = :learningGoalId
                AND p.user.id = :userId
            """)
```

Reply:
please indent the left joins and the logical operations below the "where" statement, so that the query is easier to read. Also use shorter table variables (instead of p, you can also use lgp e.g. in this case)